### PR TITLE
Add integration test for installing pysnooper

### PR DESCRIPTION
**Pull request by awarecheetah**
#39: Add tox environment for installing pysnooper.

### CODE REVIEW

**Comment by cool-RR** [comment]
Cool, can you please add a line that at least imports pysnooper to ensure that it's installed?

**Comment by awarecheetah** [comment]
> Cool, can you please add a line that at least imports pysnooper to ensure that it's installed?

Yes, I will do that with `pip show pysnooper`, which will verify the package name and version.

**Comment by alexmojaki** [comment]
You need a line with `TOXENV=pysnooper` in travis.yml. Don't know what the right stage would be, I guess test?

**Comment by alexmojaki** [comment]
I don't know why but the logs don't show the desired commands being run: https://travis-ci.org/cool-RR/PySnooper/jobs/526204104

File: tox.ini [hunk 1]
```diff
@@ -63,3 +72,8 @@ exclude = .tox,build,dist,pysnooper.egg-info
 
 [pytest]
 addopts = --strict
+
+[pysnooper]
+commands =
+    pip install pysnooper
+    pip show pysnooper
```
Comment (bittner) [inline tox.ini]: This is unneeded. You can remove it. (Explanation: The `[<tool_name>]` sections are configuration file sections that the tool reads. I don't think pysnooper reads anything out from `tox.ini`.)

File: tox.ini [hunk 2]
```diff
@@ -54,6 +56,13 @@ deps = pip-tools
 commands = pip-compile --output-file requirements.txt requirements.in
 changedir = {toxinidir}
 
+[testenv:pysnooper]
+description = Ensure pip installs pysnooper
+deps = pip
```
Comment (bittner) [inline tox.ini]: No need to install `pip`. Replace this by `pysnooper` if you want to install it off PyPI (that's probably _not_ what you want), or leave it empty.

File: tox.ini [hunk 3]
```diff
@@ -54,6 +56,13 @@ deps = pip-tools
...
+commands =
+    pip install pysnooper
+    pip show pysnooper
```
Comment (bittner) [inline tox.ini]: This is not an integration test.

Think: What does this verify? Is the pysnooper code run with it? (It's not.)

Also, installing pysnooper is unneeded, and even if it were you can move it to `deps = `.

File: tox.ini [hunk 4]
```diff
@@ -16,6 +17,7 @@ description = Unit tests
 deps =
     pytest
     python_toolbox
+    pysnooper
```
Comment (bittner) [inline tox.ini]: This installs the latest released version of pysnooper off PyPI. This is most probably _not_ what you want.

Note that pysnooper (off the checked out commit) is already installed by Tox automatically. Try running `pysnooper` or `python -c "import pysnooper; print(pysnooper.__version__)"` to prove it.

File: .travis.yml [hunk 5]
```diff
@@ -33,6 +33,14 @@ jobs:
...
   - { stage: lint, python: 3.7, env: TOXENV=bandit }
   - { stage: lint, python: 3.7, env: TOXENV=readme }
+  - { stage: test, python: 3.7, env: TOKENV=pysnooper }
+  - { stage: test, python: 3.6, env: TOKENV=pysnooper }
+  - { stage: test, python: 3.5, env: TOKENV=pysnooper }
+  - { stage: test, python: 3.4, env: TOKENV=pysnooper }
+  - { stage: test, python: 2.7, env: TOKENV=pysnooper }
+  - { stage: test, python: 3.8-dev, env: TOKENV=pysnooper }
+  - { stage: test, python: pypy2.7-6.0, env: TOKENV=pysnooper }
+  - { stage: test, python: pypy3.5, env: TOKENV=pysnooper }
```
Comment (bittner) [inline .travis.yml]: This looks very verbose.

What if you simply would run your integration test with the `[testenv]` section? (Then all would not be needed. Hmm?)

**Comment by bittner** [comment]
I think, this contribution can be much simpler:

Simply add an additional command that performs the desired integration test to the `[testenv]`'s [commands](https://github.com/cool-RR/PySnooper/blob/master/tox.ini#L19) attribute, e.g.

```ini
commands =
    pytest
    python -c "print('run my integration test')"
```

I'd also suggest to add `--verbose` [as a pytest switch](https://github.com/cool-RR/PySnooper/blob/master/tox.ini#L64-L65) (as long as you don't have hundreds of tests), e.g.
```ini
[pytest]
addopts = --strict --verbose
```
This makes the output easier to read. I see [you removed it](https://github.com/cool-RR/PySnooper/commits/master/tox.ini), Ram, due to problems with the IDE, hmm.

**Comment by bittner** [comment]
Please note, everyone, that pysnooper is _always_ installed by Tox automatically. That's the whole purpose of a Tox setup: It installs the package under test (using `setup.py`, which is always done unless [skipsdist = true](https://tox.readthedocs.io/en/latest/config.html#conf-skipsdist)).

Take a look at [older builds](https://travis-ci.org/cool-RR/PySnooper/jobs/524525195#L197-L198) to prove yourself. Note the zip (created by setuptools) and the installed pysnooper version (even though it's not specified anyhere explicitly for installation).

**Comment by awarecheetah** [comment]
Thanks for your input @bittner.  I'll make these changes as soon as I can.

**Comment by awarecheetah** [comment]
Actually, I'm just going to close out this PR.

**Pull request closed without merging**